### ISSUE TITLE
Error '[Errno 2] No such file or directory' on any requests delete or put call

### ISSUE THREAD

**Issue by vdaskalov** [ionrock/cachecontrol#155]
This problem occurs only on Python 2.7.X. On Python 3.x it is not reproducible.

So I set up a File cache like so:

```
from cachecontrol import CacheControl
from cachecontrol.caches.file_cache import FileCache

cached_session = CacheControl(session, cache=FileCache('/mydir'))
```

Afterwards if I do any delete or put request(note it's not even suppose to be through cached_session object it could be through a new requests session object) I get this:

[Errno 2] No such file or directory: '/mydir/2/1/1/b/c/211bcb5c56dcea5aaf39c4860d6ee7aeea1de598e3ef56cd0e23bd6b'

This problem does not occur if I initiate the FileCache with forever=True option.
This is a blocker to me as it prevents the proper usage of requests module.

I think the problem is in /cachecontrol/caches/file_cache.py file and around the lines:

```
try:
    FileNotFoundError
except NameError:
    # py2.X
    FileNotFoundError = IOError
```

they should rather be:

```
try:
    FileNotFoundError
except NameError:
    # py2.X
    FileNotFoundError = OSError
```

**Comment by ionrock** [comment]
Your fix was released in 0.12.3. Thanks!

**Comment by vdaskalov** [comment]
Great, thanks!
